src/runtime_src/core/tools/common/ReportCu.cpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (C) 2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include <boost/algorithm/string.hpp>
+#include "ReportCu.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+#include "core/common/utils.h"
+
+namespace qr = xrt_core::query;
+
+boost::property_tree::ptree
+parse_cu_status(uint32_t cu_status)
+{
+  boost::property_tree::ptree pt;
+  std::vector<std::string> result;
+
+  std::string cu_stat_str = xrt_core::utils::parse_cu_status(cu_status);

```

Comment:
Upon looking at this code further, it looks to be a little convoluted.  Let me explain further:
1) The cu_status value is obtained.
2) The value is passed to parse_cu_status, which parses this value and creates a string that contains each of the bit values.
3) The resulting string in step 2 is parsed and new entries are added to the property tree.

Instead, could you create a new method with a signature similar to the following:
`void parse_cu_status(uint32_t cu_status, std::vector<std::string> bit_set)`
And then iterate over this signature to populate the property tree.

**Note**
I am expecting to see the original parse_cu_status() method refactored to call the above mentioned method.